### DISCUSSION TITLE
유니티에서 코드를 분할하기 위한 방법

### DISCUSSION THREAD

**Discussion by fkdl0048** [fkdl0048/Mentoring-Project#47] [Q&A] [answered]
만약 개발 중인 시스템과 전혀 관련이 없는 미니게임을 넣어야 한다면 같은 NameSpace에 작성해야 하는건지 어셈블리를 하나 생성해서 분리하는 것이 좋은지 궁금합니다.

**Comment by jongfeel** [comment, marked as answer]
조금 더 구체적인 설명이 필요할 것 같습니다.

같은 unity project 내에 다른 scene으로 미니 게임을 개발한다는 얘기인가요?

**Reply by fkdl0048** [reply]
네네 만약 그렇게 작동하는 미니게임의 경우  NameSpace로 분리하는게 좋은지, 어셈블리단위로 분리하는 게 좋은지 궁금합니다.

++ 주고 받아야하는 데이터가 있는 경우에 어셈블리가 달라도 넘겨줄 수 있는 방법이 있을까요?

**Reply by jongfeel** [reply]
어렵게 가는 방법 보다는 그냥 쉽게 namespace로 분리하는게 좋아 보이긴 하네요.

그리고 어셈블리가 달라도 참조를 걸어 두면 별 문제 없이 쓸 수 있으므로
의존성을 분리할 수 있는 모듈인게 확실하면 어셈블리로 빼도 상관 없긴 합니다.
예) Newtonsoft.Json 같은 어셈블리는 어셈블리 모듈로 분리해서 가져와 쓰는 쪽이 익숙하기도 하고 그렇게 분리 시키는게 json이라는 객체를 컨트롤 하는데 좋기도 하므로 적절한 방법입니다.

**Reply by fkdl0048** [reply]
크게 퍼포먼스적으로 영향이 없으면 NameSpace로만 구분하거나 개수가 많아지는 것에 따라서 직접 판단하는게 중요하군요..  

감사합니다!
